### DISCUSSION TITLE
Codebase policy

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#1150] [General]
We are attempting to introduce policy documentation to help devs know what is expected and to gather up all the information spread throughout GitHub issues/PRs over time.

The basics are being introduced in https://github.com/rust-bitcoin/rust-bitcoin/pull/1127

Starting a discussion for unaddressed comments and long form discussion on policies to still to be worked out.

**Comment by tcharding** [comment]
> I'd like some rules that require more acks based on files changed or so, if we have any critical code or policies. Might be reasonably easy to add.

Posted by @Kixunil:  https://github.com/rust-bitcoin/rust-bitcoin/pull/1127#issuecomment-1198034816

**Reply by sanket1729** [reply]
Based on my experience, here is the priority of things I assign to PRs. 
1) All consensus encode/decoding code. We need to review all changes in this part of the part. The worst bug for rust-bitcoin is failing to parse a valid block or tx. This code breaks block explorers, lightning wallets that rely on rust-bitcoin etc. This should require 3 Acks or some exhaustive testing. 
2) Logical code changes to existing core APIs. In particular, these include items that almost all users of rust-bitcoin would do. For example, code that deals with amounts, addresses, bip32, bip143, and sighash computation. 2 ACKs is fine here, but this demands a more careful review or perhaps a review from an expert in that particular domain. 
3) I think two ACKs is enough for most other changes.

**Comment by tcharding** [comment]
Canonical license blurb, open under discussion still: https://github.com/rust-bitcoin/rust-bitcoin/pull/1138
